include/inviwo/core/datastructures/bitset.h

```diff
@@ -0,0 +1,399 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2021 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <inviwo/core/common/inviwocoredefine.h>
+#include <inviwo/core/util/exception.h>
+#include <inviwo/core/util/detected.h>
+
+#include <memory>
+#include <vector>
+#include <array>
+#include <iterator>
+#include <type_traits>
+#include <istream>
+#include <ostream>
+
+namespace roaring {
+class Roaring;
+class RoaringSetBitForwardIterator;
+}  // namespace roaring
+
+namespace inviwo {
+
+namespace detail {
+
+template <typename...>
+using void_t = void;
+
+template <class T, class = void>
+struct is_iterator : std::false_type {};
+
+template <class T>
+struct is_iterator<T, void_t<typename std::iterator_traits<T>::iterator_category>>
+    : std::true_type {};
+
+}  // namespace detail
+
+/**
+ * \brief represents a bitset based on roaring bitmaps provided by the CRoaring library
+ */
+class IVW_CORE_API BitSet {
```

Comment:
Should we make this serializable?

Reply:
As it is, it is just a bitset implementation more in the sense of a data source which is not yet tied to Inviwo. Might be sensible to make it also serializable in addition to readData/writeData.

Reply:
otherwise we must add it to serialize.h/deserialize.h I would rather not do that if possible.